src/classes/EntitySqlBuilder.php

```diff
@@ -55,165 +115,280 @@ public function getReadSqlBeforeWhere(bool $getTags = true, bool $fullSelect = f
-        // admin will see the experiments with visibility user for user of their team
-        if ($this->entity->Users->isAdmin) {
-            $sql .= 'AND entity.userid = users2teams.users_id)';
-        } else {
-            $sql .= 'AND entity.userid = :userid)';
+
+        $this->joinsSql[] = 'LEFT JOIN (
+                SELECT item_id AS up_item_id,
+                    (item_id IS NOT NULL) AS has_attachment,
+                    type
+                    ' . $searchAttachments . '
+                FROM uploads
+                GROUP BY item_id, type
+            ) AS uploads
+                ON (uploads.up_item_id = entity.id
+                    AND uploads.type = \'%1$s\')';
+    }
+
+    private function links(): void
+    {
+        $this->joinsSql[] = 'LEFT JOIN %1$s_links AS linkst
+            ON (linkst.item_id = entity.id)';
+    }
+
+    private function steps(): void
+    {
+        $this->selectSql[] = "SUBSTRING_INDEX(GROUP_CONCAT(
+                stepst.next_step
+                ORDER BY steps_ordering, steps_id
+                SEPARATOR '|'
+            ), '|', 1) AS next_step";
+        $this->joinsSql[] = 'LEFT JOIN (
+                SELECT item_id AS steps_item_id,
+                    body AS next_step,
+                    ordering AS steps_ordering,
+                    id AS steps_id,
+                    finished AS finished
+                FROM %1$s_steps
+                WHERE finished = 0
+            ) AS stepst
+                ON (stepst.steps_item_id = entity.id)';
+    }
+
+    private function comments(): void
+    {
+        $this->selectSql[] = 'commentst.recent_comment,
+            (commentst.recent_comment IS NOT NULL) AS has_comment';
+        $this->joinsSql[] = 'LEFT JOIN (
+                SELECT MAX(created_at) AS recent_comment,
+                    item_id
+                FROM %1$s_comments
+                GROUP BY item_id
+            ) AS commentst
+                ON (commentst.item_id = entity.id)';
+    }
+
+    /**
+     * anon filter
+     */
+    private function canAnon(string $can): string
+    {
+        return sprintf(
+            "entity.%s->'$.base' = %s",
+            $can,
+            BasePermissions::Full->value,
+        );
+    }
+
+    /**
+     * base pub/org filter
+     * scope_x is not a permission but rather a user choice to limit what is shown
+     */
+    private function canBasePubOrg(string $can): string
+    {
+        $sql = sprintf(
+            '(entity.%1$s->\'$.base\' = %2$d
+                OR entity.%1$s->\'$.base\' = %3$d)',
+            $can,
+            BasePermissions::Full->value,
+            BasePermissions::Organization->value,
+        );
+        if ($this->entity->Users->userData['scope_' . $this->entity->type] !== Scope::Everything->value) {
```

Comment:
I think this should not be in here as it is not a permission but rather a user choice to limit what is shown